File: Data Structures & Algorithms/combination-target-sum/submission-2.cpp
class Solution {
public:
    vector<vector<int>> combinationSum(vector<int>& nums, int target) {
        vector<vector<int>> res;
        vector<int> set;
        sort(nums.begin(), nums.end());
        function<void(int, int)> solve = [&](int idx, int rem) ->void {
            if (rem == 0) {
                res.push_back(set);
                return;
            }
            if (rem < 0) {
                return;
            }
            for (int i = idx; i < nums.size(); i++) {
                set.push_back(nums[i]);
                solve(i, rem - nums[i]);
                set.pop_back();
            }
        };
        solve(0, target);
        return res;
    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        n = len(nums)
        if n < 2:
            return n
        nums.sort()
        res = 1
        i = 0
        streak = 1
        while i + 1 < len(nums):
            if nums[i + 1] == nums[i]:
                i += 1
                continue
            if nums[i + 1] == nums[i] + 1:
                streak += 1
                res = max(res, streak)
            else:
                streak = 1
            i += 1
        return res

